crates/net/network/src/discovery.rs

```diff
@@ -138,6 +138,11 @@ impl Discovery {
         }
     }
 
+    /// Returns a shared reference to the discv4.
+    pub fn discv4(&self) -> Option<Discv4> {
+        self.discv4.clone()
```

Comment:
It should be thread safe to clone discv4:

```rust
/// The Discv4 frontend
///
/// This communicates with the [Discv4Service] by sending commands over a channel.
///
/// See also [Discv4::spawn]
#[derive(Debug, Clone)]
pub struct Discv4 {
    /// The address of the udp socket
    local_addr: SocketAddr,
    /// channel to send commands over to the service
    to_service: mpsc::UnboundedSender<Discv4Command>,
    /// Tracks the local node record.
    ///
    /// This includes the currently tracked external IP address of the node.
    node_record: Arc<Mutex<NodeRecord>>,
}
```

Reply:
yep this is intended for cloning